src/sentry/tasks/auto_source_code_config.py

```diff
@@ -82,8 +84,8 @@ def process_error(error: ApiError, extra: dict[str, str]) -> None:
 
 
 @instrumented_task(
-    name="sentry.tasks.derive_code_mappings.derive_code_mappings",
-    queue="derive_code_mappings",
+    name="sentry.tasks.auto_source_code_config.derive_code_mappings",
```

Comment:
I believe this name needs to remain constant as well

Reply:
You can't remove/change the `name` of a task if there are any spawn calls left in the code that is currently in production, as old web/workers will be spawning the old task, and that new task cannot be handled by new pods.

Instead, you should leave add the new task definition. You'll also need to retain the existing `instrumented_task` decorator and function signature. The 'old' function can call the new function to not duplicate logic.

If you want to avoid any task drops, you'd also need to not call the new task in these changes, as a new worker could spawn tasks that cannot be handled by an 'old' worker. Instead you could deploy the new task, and then separately deploy changes that start using the new task name.

Reply:
Thank you both! I will look into it.